crates/test-utils/src/network.rs

```diff
@@ -201,6 +201,64 @@ impl TestCluster {
         .expect("Timed out waiting for cluster to target epoch")
     }
 
+    /// Ask 2f+1 validators to close epoch actively, and wait for the entire network to reach the next
+    /// epoch. This requires waiting for both the fullnode and all validators to reach the next epoch.
+    pub async fn trigger_reconfiguration(&self) {
+        info!("Starting reconfiguration");
+        let start = Instant::now();
+
+        // Close epoch on 2f+1 validators.
+        let cur_committee = self
+            .fullnode_handle
+            .sui_node
+            .state()
+            .clone_committee_for_testing();
+        let mut cur_stake = 0;
+        for handle in self.swarm.validator_node_handles() {
+            handle
+                .with_async(|node| async {
+                    node.close_epoch_for_testing().await.unwrap();
+                    cur_stake += cur_committee.weight(&node.state().name);
+                })
+                .await;
+            if cur_stake >= cur_committee.quorum_threshold() {
+                break;
+            }
+        }
+        info!("close_epoch complete after {:?}", start.elapsed());
+
+        self.wait_for_epoch(Some(cur_committee.epoch + 1)).await;
+        self.wait_for_epoch_all_validators(cur_committee.epoch + 1)
+            .await;
+
+        info!("reconfiguration complete after {:?}", start.elapsed());
+    }
+
+    pub async fn wait_for_epoch_all_validators(&self, target_epoch: EpochId) {
```

Comment:
There's a `wait_for_nodes_transition_to_epoch` at the end of this file. Can we use that or combine these two?

Reply:
Eventually yes, but I think there are a few other calls that I couldn't get rid of just yet